js/memory.js

```diff
@@ -0,0 +1,142 @@
+		};
+
+		Card.prototype.createEl = function() {
+			var thisCard = this;
+			this.cardEl = $('<div>')
+				.addClass('each-card')
+				.data('background-color', this.color)
+				.on('click', function () {
+					thisCard.showCard();
+					if (cardAlreadyShowing) {
+						thisCard.checkForMatch(thisCard);
+					} else {
+						cardAlreadyShowing = thisCard;
+					}
+				});
+			return this.cardEl;
+		};
+
+		Card.prototype.showCard = function() {
+			this.cardEl.css('background-color', this.cardEl.data('background-color'));
+		};
+
+		Card.prototype.hideCard = function() {
+			this.cardEl.css('background-color', '#ccc');
+		};
+
+		Card.prototype.removeFromPlay = function() {
+			this.cardEl.addClass('out-of-play');
+		};
+
+		Card.prototype.checkForMatch = function(thisCard) {
+			$boardEl.addClass('checking'); // disable clicks while processing
+			if (thisCard !== cardAlreadyShowing && thisCard.color === cardAlreadyShowing.color) {
+				/* if they clicked on two (different) cards of the same color */
+				this.handleMatch(thisCard);
+			} else {
+				this.handleMismatch(thisCard);
+			}
+			totalTries++;
+			if ($('.out-of-play').length === $('.each-card').length) {
+				alert('Congratulations, you won! It took you ' + totalTries + ' tries.');
+			}
+		};
+
+		Card.prototype.handleMatch = function(thisCard) {
+			alert('Hooray, you found a match!');
+			thisCard.removeFromPlay();
+			cardAlreadyShowing.removeFromPlay();
+			cardAlreadyShowing = null;
+			$boardEl.removeClass('checking');
+		};
+
+		Card.prototype.handleMismatch = function(thisCard) {
+			/* show colors for one second, then hide again */
+			setTimeout(function() {
+				thisCard.hideCard();
+				cardAlreadyShowing.hideCard();
+				cardAlreadyShowing = null;
+				$boardEl.removeClass('checking');
+			}, 1000);
+		};
+
+		/**
+		 * Deck constructor
+		 *
+		 * @constructor
+		 * @param {number} totalCardCount The number of cards to put in the deck
+		 */
+		var Deck = function(totalCardCount) {
+			this.cards = [];
+			this.totalCardCount = totalCardCount;
+		};
+
+		Deck.prototype.build = function() {
+			var color = this.getNextColor();
+			for (var i = 0; i < this.totalCardCount; i++) {
+				if (i % 2 === 0) {
+					color = this.getNextColor();
+				}
+				this.cards.push(new Card(color));
+			}
+		};
+
+		Deck.prototype.shuffle = function() {
+			this.cards = this.cards.sort(function() {
+				return 0.5 - Math.random();
+			});
+		};
+
+		Deck.prototype.getNextColor = function() {
+			var redVal = Math.floor(Math.random() * 256);
+			var greenVal = Math.floor(Math.random() * 256);
+			var blueVal = Math.floor(Math.random() * 256);
+			return 'rgba(' + redVal + ',' + greenVal + ',' + blueVal + ', 0.7)';
+		};
+
+		/**
+		 * Board constructor
+		 *
+		 * @constructor
+		 * @param {number} size The number of cards on one side of the square
+		 */
+		var Board = function(size) {
+			this.size = size;
+			this.cardsShown = [];
+			this.totalCardCount = this.size * this.size;
```

Comment:
In general, I wouldn't store a property that can be re-computed from other data. In other words, I wouldn't store the `size` *and* the `totalCardCount`. Make one of them a method.